src/Microsoft.AspNet.Mvc.DataAnnotations/IAttributeAdapter.cs

```diff
@@ -0,0 +1,19 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNet.Mvc.ModelBinding.Validation
+{
+    /// <summary>
+    /// Interface so that adapters provide their relevent values to error messages.
+    /// </summary>
+    public interface IAttributeAdapter : IClientModelValidator
+    {
+        /// <summary>
+        /// Implementors should provide the important values of their attribute to the base
+        /// <see cref="GetErrorMessage(ModelValidationContextBase)"/>.
```

Comment:
This information references itself and assumes implementors always subclass `ValidationAttributeAdapter<TAttribute>`. Describe what this method is _for_ and say nothing about how it should be implemented e.g. `Gets the error message.`